checksum.go

```diff
@@ -190,7 +191,7 @@ func newChecksumFromValue(checksumValue, filename string) (*FileChecksum, error)
 //
 // ChecksumFromFile will only return checksums for files that match file
 // behind src
-func (c *Client) ChecksumFromFile(checksumFile string, src *url.URL) (*FileChecksum, error) {
+func (c *Client) ChecksumFromFile(ctx context.Context, checksumFile string, src *url.URL) (*FileChecksum, error) {
```

Comment:
Not sure if I follow this function well.  A couple of questions:
* does this function need to be public?  The rest are private ones
* `checksumFile` here is really a go-getter-able address and not a simple path to a local file?  Maybe rename it to `checksumURL` ?
* `src` here isn't really a source for anything relevant to function?  Maybe use `u` like `extractChecksum`?

Reply:
* I originally made this private but Packer needs it, and the context can be found here: https://github.com/hashicorp/go-getter/pull/188 , TLDR: we checksum a local file to see wether it needs to be uploaded somewhere.
* correct. I agree, changing this to `checksumURL`
* src is the src of the file being checksummed, checksums in files have a filename attached to it and this filename should match the url being downloaded. The fact that it is an url make it a little complex and only the Path part was used, changing it to a string; also I will rename it to checksummedURL, I hope this makes better sense; tell me what you think.

Reply:
( Do you have thoughts on number 3 ? )

Reply:
`checksummedURL` makes sense here - Thanks!  Thanks for the context.